demo/lightning/checkpoint/multinode/train.py

```diff
@@ -0,0 +1,236 @@
+                    f" attempt {lookup_attempt}, retrying...")
+                lookup_attempt += 1
+                time.sleep(5)
+    print(f"Coordinator IP address: {coordinator_ip_address}")
+    os.environ["MASTER_ADDR"] = str(coordinator_ip_address)
+
+
+def init_processes():
+    """Initializes the distributed environment."""
+    # Get the necessary environment variables from the GKE environment
+    world_size = int(os.environ["WORLD_SIZE"])
+
+    job_index = int(os.environ.get("JOB_INDEX"))
+    job_completion_index = int(os.environ.get("JOB_COMPLETION_INDEX"))
+    processes_in_job = int(os.environ.get("PROCESSES_IN_JOB"))
+    rank = job_index * processes_in_job + job_completion_index
+    os.environ["NODE_RANK"] = str(rank)
+
+    configure_master_addr()
+
+
+def main(project: str, ckpt_dir_path: str, save_only_latest: bool):
+    if os.environ.get("COORDINATOR_ADDRESS"):
+        init_processes()
+    dataset = WikiText2()
+    dataloader = DataLoader(dataset, num_workers=1)
+
+    model = DemoTransformer(vocab_size=dataset.vocab_size,
+                            nlayers=int(os.environ.get("NUM_LAYERS", 2)))
+    # Save once per step, and if `save_only_latest`, replace the last checkpoint each time.
+    # Replacing is implemented by saving the new checkpoint, and then deleting the previous one.
+    # If `save_only_latest` is False, a new checkpoint is created for each step.
+    checkpoint_callback = ModelCheckpoint(
+        save_top_k=1 if save_only_latest else -1,
+        every_n_train_steps=1,
+        filename="checkpoint-{epoch:02d}-{step:02d}",
+        enable_version_counter=True,
+    )
+    accelerator = os.environ.get("ACCELERATOR", "gpu")
+    trainer = Trainer(default_root_dir=ckpt_dir_path,
+                      plugins=[],
+                      callbacks=[checkpoint_callback],
+                      min_epochs=4,
+                      max_epochs=5,
+                      max_steps=3,
+                      accelerator=accelerator,
+                      strategy=DatafluxFSDPStrategy(
+                          path=ckpt_dir_path,
+                          project_name=project,
+                          storage_client=None,
+                          model=model,
+                          state_dict_type="sharded",
+                      ),
+                      num_nodes=1
+                      )
+    trainer.fit(model, dataloader)
+    new_path = ckpt_dir_path + \
+        "lightning_logs/version_0/checkpoints/checkpoint-epoch=00-step=03.ckpt"
+    print("Restoring checkpoints ...")
+    model = DemoTransformer(vocab_size=dataset.vocab_size,
+                            nlayers=int(os.environ.get("NUM_LAYERS", 2)))
+    trainer = Trainer(default_root_dir=ckpt_dir_path,
+                      plugins=[],
+                      callbacks=[],
+                      min_epochs=4,
+                      max_epochs=5,
+                      max_steps=3,
```

Comment:
These should probably all be configurable

Reply:
Ack will do this in a follow up PR. Filed b/366046586